examples/by_feature/checkpointing.py

```diff
@@ -200,6 +200,8 @@ def training_function(config, args):
                 optimizer.step()
                 lr_scheduler.step()
                 optimizer.zero_grad()
+            if isinstance(checkpointing_steps, int):
+                overall_step += 1
```

Comment:
Why not always do this?

Reply:
It's only needed if we save via checkpointing steps rather than epoch, so didn't want people to assume we *always* need to do that. 

(Which means a comment is needed!)

Reply:
Yeah I get the variable won't be used if we don't checkpoint with steps, but it doesn't hurt to always have it (and would save one line of code).